op-service/eth/sync_status.go

```diff
@@ -3,17 +3,14 @@ package eth
 // SyncStatus is a snapshot of the driver.
 // Values may be zeroed if not yet initialized.
 type SyncStatus struct {
-	// CurrentL1 is the L1 block that the derivation process is currently at in the inner-most stage.
+	// CurrentL1 is the L1 block that the derivation process is last idled at.
```

Comment:
Previously it was guaranteed that the block _before_ this one was fully reflected in the node's safe head.  I think if anything we've made this stricter but it's an important condition for the challenger to know when it's safe to use data from the node to play games so important we don't weaken it.

Reply:
Hmm, we can add a `DerivedSafe` and `DerivedFrom` attribute to the sync-status for the safe-block, that is then much more precise/safe. Since the safe-block just reflects the engine, which is not atomically coupled to the derivation. I'll open an issue for that. For now I'll update the `CurrentL1` behavior to update with every single L1 block that the end of the pipeline reports, not just when idle, to ensure it's always as up to date as possible.

Reply:
Opened issue: https://github.com/ethereum-optimism/optimism/issues/11014